### PR TITLE
feat: Add intuitive closing logic for flyout

### LINKED ISSUE #3319: PopupManager needs intuitive closing logic

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3319

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/layout/src/Modal.ts [hunk 1]
```diff
@@ -86,12 +89,17 @@ export class Modal extends HTMLWidget {
...
                 ;
         } else if (this.fixedHeight() !== null && this.fixedWidth() !== null) {
+            console.log("here1");
             this._modal
-                .style("top", (rect.top + (rect.height / 2) - (modalRect.height / 2)) + "px")
-                .style("left", (rect.left + (rect.width / 2) - (modalRect.width / 2)) + "px")
-                ;
+            .style("top", (rect.top + (rect.height / 2) - (modalRect.height / 2)) + "px")
+            .style("left", (rect.left + (rect.width / 2) - (modalRect.width / 2)) + "px")
+            ;
```
Comment (GordonSmith) [inline packages/layout/src/Modal.ts]: Lost indentation (whitespace change?)

File: packages/layout/src/Modal.ts [hunk 2]
```diff
@@ -86,12 +89,17 @@ export class Modal extends HTMLWidget {
...
         } else if (this.minHeight() || this.minWidth()) {
+            console.log("here2");
+            console.log("rect.top", rect.top);
+            console.log("rect.height/2", rect.height / 2);
             const contentRect = this._modal.node().getBoundingClientRect();
+            console.log("contentRect.height/2", contentRect.height / 2);
```
Comment (GordonSmith) [inline packages/layout/src/Modal.ts]: Remove console.log

File: packages/layout/src/Modal.ts [hunk 3]
```diff
@@ -167,6 +175,7 @@ export class Modal extends HTMLWidget {
 
         this.setModalSize();
         const rect = this._relativeTarget.getBoundingClientRect();
+        console.log("this._relativeTarget", this._relativeTarget);
```
Comment (GordonSmith) [inline packages/layout/src/Modal.ts]: remove console.log

**Comment by GordonSmith** [comment]
Looking at the screen shot - it appears the modal window is way bigger than the form?

**Comment by GordonSmith** [comment]
@jbrundage bump.

**Comment by jbrundage** [comment]
It uses the same min height and width as the previous popup. The size of the FieldForm isn't available until after it renders. We could approximate the FieldForm height and width, or we could render it twice, transparent the first time and then render it at the correct size immediately afterwards? I'll explore these options.

**Comment by jbrundage** [comment]
@GordonSmith I've amended this PR with your suggested changes. And I've added logic to resize the modal to fit its content in the callback of the render method within the filter popup click handler.

**Comment by jbrundage** [comment]
image

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/Visualization/pull/3392 (minor tweaks - but your implementation works and looks great).

**Pull request closed without merging**